app/scripts/lib/multichain-api/provider-authorize/helpers.ts

```diff
@@ -0,0 +1,61 @@
+import { CaipAccountId, Hex } from '@metamask/utils';
+import {
+  NetworkClientId,
+  NetworkController,
+} from '@metamask/network-controller';
+import { ScopesObject, validateScopedPropertyEip3085 } from '../scope';
+
+export const assignAccountsToScopes = (
+  scopes: ScopesObject,
+  accounts: Hex[],
+) => {
+  Object.keys(scopes).forEach((scope) => {
+    if (scope !== 'wallet') {
+      scopes[scope].accounts = accounts.map(
+        (account) => `${scope}:${account}` as unknown as CaipAccountId, // do we need checks here?
+      );
+    }
+  });
+};
+
+export const validateAndUpsertEip3085 = async ({
+  scopeString,
+  eip3085Params,
+  origin,
+  upsertNetworkConfiguration,
+  findNetworkClientIdByChainId,
+}: {
+  scopeString: string;
+  eip3085Params: unknown;
+  origin: string;
+  upsertNetworkConfiguration: NetworkController['upsertNetworkConfiguration'];
+  findNetworkClientIdByChainId: NetworkController['findNetworkClientIdByChainId'];
+}): Promise<undefined | NetworkClientId> => {
+  const validParams = validateScopedPropertyEip3085(scopeString, eip3085Params);
+
+  const {
+    chainId,
+    chainName,
+    firstValidBlockExplorerUrl,
+    firstValidRPCUrl,
+    ticker,
+  } = validParams;
+
+  try {
+    findNetworkClientIdByChainId(chainId as Hex);
+    return undefined;
+  } catch (err) {
+    // noop
+  }
```

Comment:
is this still required? seems like we're doing this earlier in the request now?

Reply:
early in the request we are checking that a scope _can_ be served by either a current network client or a future network client added via 3085 in the scopedProperties.

Here we check if the particular 3085 should be added or not since this helper is called in a loop from provider_authorize on all valid 3085 params